website_sale_product_minimal_price/models/product_template.py

```diff
@@ -12,12 +14,31 @@ class ProductTemplate(models.Model):
     )
 
     def _compute_has_distinct_variant_price(self):
+        partner = self.env.context.get('partner')
+        pricelist_id = self.env.context.get('pricelist')
         for template in self:
             if template.product_variant_count > 1:
-                prices = template.product_variant_ids.mapped('website_price')
-                if len(prices) > 1:
+                prices = template._get_variants_price(pricelist_id, 1, partner)
+                if len(set(prices.values())) > 1:
                     template.has_distinct_variant_price = True
 
+    # TODO: Use ormcache but we need invalidate cache in all models involved
+    #  in a price change
+    @ormcache('self.id', 'pricelist_id', 'qty', 'partner')
```

Comment:
You can use `ormcache_context` and remove `pricelist_id` and `partner` arguments if you prefer, because AFAICS those context keys are always used to obtain the variants price.

```suggestion
    @ormcache_context('self.id', 'qty', keys=('partner', 'pricelist'))
```

In any case, I would consider also adding `self.uid` as key, because I guess different users would have different access levels and could see different pricelists, etc. In general it's a good idea to cache per user:

```suggestion
    @ormcache('self.id', 'self.uid', 'pricelist_id', 'qty', 'partner')
```

Both ideas together:


```suggestion
    @ormcache_context('self.id', 'self.uid', 'qty', keys=('partner', 'pricelist'))
```

Choose what you prefer :blush: